pkg/odo/cli/component/common_push.go

```diff
@@ -330,3 +332,75 @@ func filterIgnores(filesChanged, filesDeleted, absIgnoreRules []string) (filesCh
 	}
 	return filesChangedFiltered, filesDeletedFiltered
 }
+
+// retrieveDefaultComponentNamespace tries to retrieve the current active namespace
+// to set as a default namespace
+func retrieveKubernetesDefaultNamespace() (string, error) {
+	// Get current active namespace
+	client, err := kclient.New()
+	if err != nil {
+		return "", err
+	}
+	return client.Namespace, nil
+}
+
+// checkNamespaceDefaultIfNotAvailable checks to see if a namespace has been passed in, if not, we will default
+// to the standard kubernetes namespace.
+func checkCmdNamespaceDefaultIfNotAvailable(cmd *cobra.Command) (string, error) {
```

Comment:
I am not sure here, but we have similar functions in context.go.
is it possible to reuse them here??
https://github.com/openshift/odo/blob/77c876beef3f456067bff41b727914c6bf9a3725/pkg/odo/genericclioptions/context.go#L311
https://github.com/openshift/odo/blob/77c876beef3f456067bff41b727914c6bf9a3725/pkg/odo/genericclioptions/context.go#L343

Reply:
Ah sorry, I decided to create a different function because the functionality is slightly different. We check the two flags and then default to the standard kubernetes namespace if available. Similar to the code that I had deleted / overridden as a result of this PR.